src/server/pkg/collection/collection.go

```diff
@@ -397,6 +406,33 @@ func (c *readonlyCollection) GetBlock(key string, val proto.Unmarshaler) error {
 	}
 }
 
+func endKeyFromPrefix(prefix string) string {
+	// Lexigraphically increment the last character
+	return prefix[0:len(prefix)-1] + string(byte(prefix[len(prefix)-1])+1)
```

Comment:
What happens if the last character is the maximum value here so incrementing it leads to wrap around behavior? Is that something we need to worry about?

Reply:
Realistically, I think we're ok.

Since go uses UTF-8 the maximum value would be the `\xff` value (which is a non printable character).

As long as users name their files sane things (within the ASCII range), even if they use the max printable ASCII character (which is `~` or `\x7e`), incrementing this value is a valid byte that will serve as a valid endKey.

There is however, the matter of what a user chooses to name a file. Right now we only make sure the null character isn't in the file path. So in theory a user could use non printable UTF-8 bytes in filenames, and if they supplied `\xff` then yea, this would overflow and cause a problem. [We should probably just validate file paths to be printable ASCII characters](https://github.com/pachyderm/pachyderm/issues/2614)

Reply:
This seems reasonable.